### PR TITLE
Add full-package test fixtures & integration tests

### PR DESCRIPTION
**Pull request by FredKSchott**
Currently, our current checked-in test fixtures ("case-map", "polymer-element") were small and somewhat broken. Testing of the full "polymer" element was opt-in and impossible to test via travis (since expected output was never checked in). While we test individual conversion points in our unit tests, we don't currently have any automated integration tests on real-world packages.

---

This PR adds the ability to test full package conversion via the top-level, CLI interface. These three fixtures are ready to be added (in [a separate PR](227) due to the # of files):

1. @polymer/polymer
1. @polymer/iron-icon
1. @polymer/paper-button

--- 

Together, these conversions & tests take ~30 seconds to run. To protect all of our sanity these integration tests are not run by `npm test` or `npm run test:watch`. You can run them manually via `npm run test:full`, or wait for Travis to run them and block your PR if there are any issues.

```
  "test:full": "npm run build && mocha lib/test/*_test.js lib/test/**/*_test.js && npm run lint",
  "test:fast": "npm run build && mocha lib/test/unit/*_test.js lib/test/unit/**/*_test.js",
  "test": "npm run test:fast && npm run lint",
```

--- 

Fixtures are checked in to enable deterministic CI testing. Each fixture directory has the following format:

- `fixtures/packages/PACKAGE_NAME`:
  - `/source/`: Checked in, Created by `npm run build:fixtures`
  - `/expected/`: Checked in, Created by `npm run build:fixtures`
  - `/generated/`: Created & compared by `npm run test:full`
  - `/test.js`: Checked in, per-package test configuration

---

┻┳| 
┳┻| _ 
┻┳| •.•)   The 3 test fixtures made this PR too big, 
┳┻|⊂ﾉ       so they're added in [a separate PR](227)
┻┳|

### CODE REVIEW

File: fixtures/packages/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Place test fixtures here!
+
+Fixtures are checked in to enable deterministic CI testing. Each fixture directory has the following format: 
+
+- `fixtures/packages/PACKAGE_NAME`: 
+  - `/source/`: Checked in, Created by `npm run build:fixtures` 
+  - `/expected/`: Checked in, Created by `npm run build:fixtures` 
+  - `/generated/`: Created & compared by `npm run test:full` 
+  - `/test.js`: Checked in, per-package test configuration
```
Comment (usergenic) [inline fixtures/packages/README.md]: Maybe note that `fixtures/packages/PACKAGE_NAME/test.js` this is the only asset for any test which is created/maintained by hand.

File: src/test/fixtures/fixtures_test.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+import * as chalk from 'chalk';
+import * as diff from 'diff';
+import * as fs from 'mz/fs';
+import * as path from 'path';
+import * as rimraf from 'rimraf';
+
+import exec from '../util/exec';
+
+// Install source map support for stack traces, etc.
+require('source-map-support').install();
```
Comment (usergenic) [inline src/test/fixtures/fixtures_test.ts]: You can also put `--require source-map-support/register` as a line in `mocha.opts` to achieve this and it ensures it is loaded as early as possible so you can ensure proper .ts file references in places which might be loaded early; for example `../util/exec.ts` lines might not be mapped?  I'd have to check that bit ...

Comment (FredKSchott) [reply src/test/fixtures/fixtures_test.ts]: Ha, you know what it's already in there. Removing this then...

File: fixtures/packages/README.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Place test fixtures here!
+
+Fixtures are checked in to enable deterministic CI testing. Each fixture directory has the following format: 
+
+- `fixtures/packages/PACKAGE_NAME`:
```
[the remark this answers is not in the document]
Comment (FredKSchott) [reply fixtures/packages/README.md]: No they don't, they match the "package" layout where `bower_components/` live inside the package repo.

A future PR will add a workspace fixture where all packages resolve against each other.

**Comment by FredKSchott** [comment]
@justinfagnani @usergenic PTAL

**Pull request merged**